src/analysis/function_analysis.h

```diff
@@ -50,60 +51,61 @@ class LivenessAnalysis {
     // we don't keep track of node->parent_block relationships, so you have to pass both:
     bool isKill(AST_Name* node, CFGBlock* parent_block);
 
-    bool isLiveAtEnd(InternedString name, CFGBlock* block);
+    bool isLiveAtEnd(int vreg, CFGBlock* block);
 };
 
 class PhiAnalysis;
 
 class DefinednessAnalysis {
 public:
     enum DefinitionLevel {
+        Unknown,
```

Comment:
I think now that this VRegMap is an array it could save a lot of memory if we use `enum DefinitionLevel : char`